logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/SystemMessageBuilder.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.data.message
+
+import com.benasher44.uuid.uuid4
+import com.wire.kalium.logic.data.event.Event
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.util.DateTimeUtil
+
+interface SystemMessageBuilder {
```

Comment:
The name `Builder` would make me think of the Builder pattern, and not a class that creates & persists messages.

Are there plans to use this class anywhere else other than in `ProtocolUpdateEventHandler`? If not, we could just add some utilities around that class.

Edit: I see it's also used in `MLSMigrator`, and it makes a ton of sense.

Perhaps just renaming it?

```suggestion
interface SystemMessagesManager {
```

```suggestion
interface ProtocolMigrationMessagesManager {
```

Reply:
The plan was to collect the construction of system message in one place since the same system message is often inserted from several code paths.

I'm happy to rename it.

Reply:
Renamed it to `SystemMessageInserter`